Important: #95647 - Composer installations and extension usage (#1585)
TYPO3-Documentation/TYPO3CMS-Reference-CoreApi: Documentation/ExtensionArchitecture/FilesAndLocations/Index.rst

is placed at :file:`Resources/Public/Icons/Extension.svg`, the Extension Manager
will show that image.

Most of these files are not required. The exception are :file:`ext_emconf.php`

> in legacy (non-Composer) installations and :file:`composer.json` in

Reviewer: ```suggestion
in Legacy installations and :file:`composer.json` in
```

> Composer-based installations.

Reviewer: ```suggestion
Composer installations.
```

.. note::
   It is recommended to keep :file:`ext_emconf.php` and :file:`composer.json` in
   any public extension that is published to TYPO3 Extension Repository (TER), and

>    to ensure optimal compatibility with Composer-based installations and legacy
>    mode.

Reviewer: ```suggestion
   to ensure optimal compatibility with Composer and Legacy installations.
```

:file:`composer.json`
---------------------

> *-- required* in composer-based installations

Reviewer: ```suggestion
*-- required* in Composer installations
```

Author: see above

For more information, see :ref:`composer-json`.

.. versionchanged:: 11.4
   The ordering of installed extensions and their dependencies are loaded from
   the :file:`composer.json` file, instead of :file:`ext_emconf.php` in

>    composer-based installations.

Reviewer: ```suggestion
   Composer installations.
```

Author: see above

.. note::
   Extension authors should ensure that the information in the :file:`composer.json`
   file is in sync with the one in the extensions' :file:`ext_emconf.php` file.
   This is especially important regarding constraints like `depends` , `conflicts`
   and `suggests`. Use the equivalent settings in :file:`composer.json` `require`,
   `conflict` and `suggest` to set dependencies and ensure a specific loading order.

.. index:: File; EXT:{extkey}/ext_emconf.php
.. _ext_emconf-php:

:file:`ext_emconf.php`
----------------------

> *-- required* in legacy installations

Reviewer: ```suggestion
*-- required* in Legacy installations
```

Author: see above

Definition of extension properties.

> Name, category, status etc. are used by the Extension Manager in legacy

Reviewer: ```suggestion
Name, category, status etc. are used by the :guilabel:`Extensions` module in Legacy
```

installations. The content of this file is described in more details in
:ref:`extension-declaration`.

> For legacy non-Composer installation the :file:`ext_emconf.php` file is the

Reviewer: ```suggestion
For Legacy installations the :file:`ext_emconf.php` file is the
```

Author: see above
